eng/SourceBuild.props

```diff
@@ -31,10 +31,12 @@
       <InnerBuildArgs>$(InnerBuildArgs) --nodereuse false</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --warnAsError false</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) --outputrid $(TargetRid)</InnerBuildArgs>
-      <!-- RuntimeOS is the build host rid OS. -->
+      <!-- PackageOS and RuntimeOS control the rids of prebuilts consumed by the build.
+           They are set to RuntimeOS so they match with the build SDK rid. -->
+      <InnerBuildArgs>$(InnerBuildArgs) /p:PackageOS=$(RuntimeOS) /p:ToolsOS=$(RuntimeOS)</InnerBuildArgs>
```

Comment:
I've eliminated `RuntimeOS` in the rid calculation because it was an overloaded term with an ambiguous name.

The only user that remains is source-build, which uses `RuntimeOS` to pass the rid of the build sdk.

I kept it local to this file to make explicit source-build is the only user.
Is that ok? Or would you prefer to implement it as an override for `PackageOS` and `ToolsOS` in `Directory.Build.props`?

Reply:
We try to follow the [GCC configure terms](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html#:~:text=So%2C%20build%20is%20the%20machine,you're%20building%20libraries) which expose `Build`, `Host` and `Target` as terms to configure a build process. It would be good to follow that consistently.

Unsure if that answers your questions but I thought it might be helpful for context.

Reply:
I don't think we capture all 3 in `Directory.Build.props`.
If I make the analogy, I'd map _GCC Build_ to `ToolsOS`, and _GCC Target_ to `PackageOS`.
Depending on the target, appropriate things are built for the _GCC Host_.

I think we can leave `RuntimeOS` in `SourceBuild.props` (just for source-build) until we have a reason to move it to `Directory.Build.props`.